src/fits_schema/schema_element.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+"""Defines common schema metadata inherited by Column and HeaderCard."""
+
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
+from astropy.units import Unit
+from astropy.io.votable.ucd import check_ucd
+
+# collects all references used, useful for generating a citation list.
+_REFERENCE_SET = set()
+
+
+@dataclass
+class SchemaElement(metaclass=ABCMeta):
+    """Base for Schema elements defining common metadata keys."""
+
+    #: name to use if not the attribute name, useful in case the column or
+    #: header has a non-python-allowed name, i.e. with a ``-`` or starting with a
+    #: number
+    name: str | None = None
+
+    #: human-readable description
+    description: str = ""
+
+    #: if this item is optional, set to false
+    required: bool = True
+
+    #: the unit that should be checked for compatibility
+    unit: Unit | None = None
```

Comment:
This is the common base-class for both `Column` and `HeaderCard`, right?

But HeaderCards do not have units

Reply:
It's mostly for documentation (i.e if you have a header called E_MAX, you want to know the unit to put there in the documentation).  The FITS standard allows you to specify a unit in square brackets in the comment string as well, so in principle we could check for that and warn if it's not what is expected.

Reply:
Probably I should change the help string then, to be clearer about that